internal/ignition/installmanifests.go

```diff
@@ -751,15 +752,22 @@ func (g *installerGenerator) updateBootstrap(ctx context.Context, bootstrapPath
 			// get corresponding host
 			var host *models.Host
 			masterHostnames := getHostnames(masters)
+			arbiterHostnames := getHostnames(arbiters)
 			workerHostnames := getHostnames(workers)
 
 			// The BMH files in the ignition are sorted according to hostname (please see the implementation in installcfg/installcfg.go).
 			// The masters and workers are also sorted by hostname.  This enables us to correlate correctly the host and the BMH file
-			if bmhIsMaster(bmh, masterHostnames, workerHostnames) {
+			role := getBmhRole(bmh, masterHostnames, arbiterHostnames, workerHostnames)
+			if role == models.HostRoleMaster {
 				if len(masters) == 0 {
 					return errors.Errorf("Not enough registered masters to match with BareMetalHosts")
 				}
 				host, masters = masters[0], masters[1:]
```

Comment:
It feels like this might be worth breaking out into a function at this point and reusing it for each role.